R/s3.get.R

```diff
@@ -1,11 +1,39 @@
 s3.get <- function (bucket, bucket.location = "US", verbose = FALSE, debug = FALSE) {
   AWS.tools:::check.bucket(bucket)
-  x.serialized <- tempfile()
-  s3.cmd <- paste("s3cmd get", bucket, x.serialized, paste("--bucket-location",
-      bucket.location), ifelse(verbose, "--verbose --progress",
-      "--no-progress"), ifelse(debug, "--debug", ""))
-  system(s3.cmd)
-  ans <- readRDS(x.serialized)
-  unlink(x.serialized)
+
+  # Helper function for fetching data from s3
+  fetch <- function(){
+    x.serialized <- tempfile()
+    s3.cmd <- paste("s3cmd get", bucket, x.serialized, paste("--bucket-location",
+    bucket.location), ifelse(verbose, "--verbose --progress",
+    "--no-progress"), ifelse(debug, "--debug", ""))
+    system(s3.cmd)
+
+    ans <- readRDS(x.serialized)
+    unlink(x.serialized)
+    ans
+  }
+
+  # Check for the bucket in the cache
+  # If it does not exist, create and return its entry
+  if (!s3cache$exists(bucket)) {
+    ans <- fetch()
+    s3cache$set(bucket, ans)
+  } else{
+    # Check time on s3cache's copy
+    last_cached <- s3cache$last_accessed(bucket) # assumes a POSIXct object
+
+    # Check time on s3 remote's copy
+    s3.cmd <- paste("s3cmd ls ", bucket, "| awk '{print $1" "$2}' ")
```

Comment:
It's cool we can worry about it later when I add [appveyor](http://www.appveyor.com/) to all the repos